### PR TITLE
pfunit: add mpi-test patch

### PR DESCRIPTION
**Pull request by junghans**
/cc @AlexisPerry  @cferenba

### CODE REVIEW

**Comment by junghans** [comment]
@cferenba Can you write a blurb of documentation for that patch please?

**Comment by cferenba** [comment]
Blurb:
"This patch modifies pfunit to be usable by codes that have MPI_COMM_WORLD hardcoded as their communicator, by skipping over any tests whose PE count does not match the size of the current MPI_COMM_WORLD."

Along with this, I'd recommend renaming the patch to `mpi_comm_world.patch`, and the variant to something like `use-comm-world`.

File: var/spack/repos/builtin/packages/pfunit/package.py [hunk 1]
```diff
@@ -41,12 +41,16 @@ class Pfunit(CMakePackage):
     variant('shared', default=True,
             description='Build shared library in addition to static')
     variant('mpi', default=False, description='Enable MPI')
+    variant('use-comm-world', default=False, description='Enable MPI_COMM_WORLD for testing')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pfunit/package.py]: I prefer variants to use underscores over dashes as the minus sign is used to disable variants, and `+use-comm-world` can seem ambiguous for new users.

Comment (junghans) [reply var/spack/repos/builtin/packages/pfunit/package.py]: Fixed!

**Pull request merged**